### PR TITLE
Recursion - Recursive Sequence

### PR DESCRIPTION
**Pull request by prateek-mehra**
## Checklist
- Solution is tested on sphere engine <br/>
    `Note:` Please copy only the solution part. Do not copy/paste the template code. That way, any issues with template code can be caught early on.
    - [ ] Author
    - [ ] Reviewer
- PR title is correct - [PR title conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#title-for-a-pull-request)
    - [ ] Author
    - [ ] Reviewer
- Folder/File naming is correct - [Naming conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#naming-of-files-and-folders)
    - [ ] Author
    - [ ] Reviewer
- Coding conventions are followed - [Coding conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#coding-conventions)
    - [ ] Author
    - [ ] Reviewer
- Test cases (action item for **reviewer**) <br/>
    - To be ensured by checking the problem on sphere-engine
    - If any test cases are missing, please add them - [how to add new test cases](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#adding-new-test-cases)
    - [ ] There are at least 2 test cases
    - [ ] There is at least 1 test case that is different from the sample test cases

### CODE REVIEW

**Comment by vinthaivendhan10x** [comment]
@LVSsandeepVarma kindly Review

**Review by LVSsandeepVarma — CHANGES_REQUESTED** [review]
@prateek-mehra  , since we are giving this problem in  recursion topic we need a recursive solution instead of iterative solution

File: solutions/recursion/recursive-sequence.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+// -------- Do NOT edit anything above this line ----------
+// Use readLine() for taking input, it will read one line of from the input  and is stored in string format
+
+function f(operands, num, n){
+
+    if(operands == n+1)
+        return 0;
+
```
[the remark this answers is not in the document]
Comment (prateek-mehra) [reply solutions/recursion/recursive-sequence.js]: 1. Yes I mentioned **f** for better understanding.

2. Done.

[the remark this answers is not in the document]
Comment (vinthaivendhan10x) [reply solutions/recursion/recursive-sequence.js]: @LVSsandeepVarma kindly Review

**Review by LVSsandeepVarma — APPROVED** [review]

File: solutions/recursion/recursive-sequence.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+function readLine() {
+	idx++;
+	return data[idx - 1];
+}
+
+// -------- Do NOT edit anything above this line ----------
+// Use readLine() for taking input, it will read one line of from the input  and is stored in string format
+
+function f(operands, num, n){
```
Comment (sailesh-ramayanam) [inline solutions/recursion/recursive-sequence.js:14]: @prateek-mehra 
1. We don't need the parameters `operands` and `num`. Just `n` is sufficient.
2. `currentTerm` will be a product of `n` integers, starting from (n*(n-1)/2)
3. This implementation can be rewritten as `currentTerm + f(n-1)`.
4. Also, please format the code properly. There are extra empty lines, extra indentation etc. If you are using VS Code, you can use the built-in formatter. Right click -> Format Document

**Review by sailesh-ramayanam — CHANGES_REQUESTED** [review]
@prateek-mehra

**Comment by vinthaivendhan10x** [comment]
@LVSsandeepVarma kindly Review

**Review by sailesh-ramayanam — APPROVED** [review]

**Pull request merged**